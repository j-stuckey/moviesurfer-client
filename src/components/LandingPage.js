import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Redirect, Link } from 'react-router-dom';

class LandingPage extends Component {
    render() {
        const { isLoggedIn } = this.props;

        return (
            <div>
                {isLoggedIn && <Redirect to="/search" />}
                <h1>Welcome to Moviesurfer!</h1>
                <p>Welcome page still in development...</p>

                <Link to="/register">Sign Up</Link>
				<Link to="/login">Login</Link>
            </div>
        );
    }
}

const mapStateToProps = state => ({
    isLoggedIn: state.auth.currentUser !== null
});

export default connect(mapStateToProps)(LandingPage);
