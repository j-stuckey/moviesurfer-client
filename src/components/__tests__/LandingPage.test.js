import React from 'react';
import { LandingPage } from '../LandingPage';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';


describe('<LandingPage />', () => {
	const wrapper = shallow(<LandingPage />);

    it('renders without crashing', () => {
        expect(wrapper.exists()).toBe(true);
    });

    it('renders correctly', () => {	
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
