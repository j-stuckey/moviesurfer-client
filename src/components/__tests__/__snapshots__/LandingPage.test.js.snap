// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LandingPage /> renders correctly 1`] = `
<div>
  <div
    id="welcome-page"
  >
    <h1>
      Welcome to Moviesurfer!
    </h1>
    <p>
      Welcome page still in development...
    </p>
    <Link
      replace={false}
      to="/register"
    >
      Sign Up
    </Link>
    <Link
      replace={false}
      to="/login"
    >
      Login
    </Link>
  </div>
</div>
`;
